.github/workflows/docker.yml

```diff
@@ -100,10 +100,41 @@ jobs:
       - id: setup_tags
         run: echo "tag=$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
 
+  setup_version_properties:
+    name: Setup version properties
+    runs-on: ubuntu-latest
+    steps:
+      - uses: RDXWorks-actions/checkout@main
+        with:
+          fetch-depth: 0
+      - id: setup_version_properties
+        run: |
+          echo "VERSION_BRANCH=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}" >> $GITHUB_ENV
+          echo "VERSION_BUILD=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "VERSION_COMMIT=${GITHUB_SHA}" >> $GITHUB_ENV
+          echo "VERSION_LAST_TAG=$(git tag --sort=-creatordate | head -n 1)" >> $GITHUB_ENV
+          if [[ $(git describe --exact-match --tag 2>&1) != *"no tag exactly matches"* ]]; then 
+            echo "VERSION_TAG=$(git describe --exact-match --tag)" >> $GITHUB_ENV
```

Comment:
As an opportunity to improve the behaviour -- this caused us quite frustrating behaviour where an old tag name was used; if we had multiple tags against the same commit.

Is there any way to read it from the triggering tag / ref instead?

e.g. If the job is caused by the creation of a release, we use that tag name, if not, it's empty?

Reply:
We'd then need to tweak `calculateVersionString` in `ApplicationVersion.java` to return `details.get("tag")` if it exists, instead of using this `cleanTag` stuff, and possibly update `ApplicationVersionTest.java`.

Reply:
Yep, I can look at this. I initially just wanted to get a like-for-like solution but happy to make improvements :)

Reply:
I've updated the github actions to use the tag that triggered a run, rather than query github for the latest tag, etc.
I've removed the `cleanTag` stuff from `calculateVersionString`.